src/components/Dialogs/Dialog/Dialog.tsx

```diff
@@ -26,20 +33,19 @@ export interface BaseDialogProps extends ModalProps {
     variant?: Omit<ButtonContext, 'secondary'>;
 }
 
-export type DialogProps = BaseDialogProps &
-    ({ title?: string; onClose: () => void } | { title?: undefined; onClose?: never });
+export type Props = BaseProps & (HeaderProps | { title?: undefined; closeButton?: undefined });
 
 const { block, elem } = bem('Dialog', styles);
 
-export const Dialog: React.FC<DialogProps> = (props) => {
+export const Dialog: React.FC<Props> = (props) => {
     const {
         isOpen,
         children,
         title,
         acceptButton,
         cancelButton,
+        closeButton,
         variant = 'primary',
-        onClose,
```

Comment:
I guess it is a breaking change, no? Let's make sure to document it.

Reply:
I see Ammar already wrote something, but it is very vague in my opinion and hard to execute when updating the application. It would be nice to write things like `onRequestClose` is replaced by `closeButton:... ` or whatever makes sense. So more like a checklist on how to migrate. What do you think?